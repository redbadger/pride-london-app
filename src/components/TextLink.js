// @flow
import React from "react";
import { View, StyleSheet } from "react-native";
import Text from "./Text";
import { eucalyptusGreenColor } from "../constants/colors";

type Props = {
  children: string
};

// Note: No touch functionality here. Instead expect to be wrapped in a
// touchable. This was due to the rouch target sizing, and the fact that
// often you want to have this and other siblings to be touchable.
const TextLink = ({ children }: Props) => (
  <View style={styles.link}>
    <Text type="h4" color="lightNavyBlueColor">
      {children}
    </Text>
  </View>
);

const styles = StyleSheet.create({
  link: {
    borderBottomColor: eucalyptusGreenColor,
    borderBottomWidth: StyleSheet.hairlineWidth * 2,
    alignSelf: "flex-start",
    marginBottom: 2
  }
});

export default TextLink;
