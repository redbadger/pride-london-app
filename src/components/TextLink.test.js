// @flow
import React from "react";
import { shallow } from "enzyme";
import TextLink from "./TextLink";

it("renders correctly", () => {
  const output = shallow(<TextLink onPress={() => {}}>Some text</TextLink>);
  expect(output).toMatchSnapshot();
});
