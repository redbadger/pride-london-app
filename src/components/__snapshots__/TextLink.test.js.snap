// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component
  style={
    Object {
      "alignSelf": "flex-start",
      "borderBottomColor": "#2cda9d",
      "borderBottomWidth": 2,
      "marginBottom": 2,
    }
  }
>
  <Text
    color="lightNavyBlueColor"
    markdown={false}
    markdownStyle={Object {}}
    type="h4"
  >
    Some text
  </Text>
</Component>
`;
